### PR TITLE
Incremental build improvements

### PR DESCRIPTION
**Pull request by markxnelson**
# Description

This PR implements continued incremental improvements for build and tests. 

- Move coverage to the top level, remove duplication, count more files in total coverage number.
- Update TPL file
- Remove dependency on verrazzano-crd-generator project, in preparation for retirement of that project.  To do this, I copied the WebLogic domain CR from 0.3.34 into this project. 
- Fix a number of fmt/lint issues
- Fix an issue in QueryMetric() which was calling ginkgo.Fail() inside an Eventually() causing only one attempt to be made, and causing tests to fail sporadically.

Contributes to VZ-2371

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [x] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: THIRD_PARTY_LICENSES.txt [hunk 1]
```diff
@@ -2247,9 +2307,11 @@ github.com/json-iterator/go
 github.com/nxadm/tail
 github.com/onsi/ginkgo
 github.com/onsi/gomega
+github.com/stretchr/testify
 go.uber.org/atomic
 go.uber.org/multierr
 go.uber.org/zap
+gopkg.in/yaml.v3
```
Comment (pfmackin) [inline THIRD_PARTY_LICENSES.txt:2314]: we already use sigs.k8s.io/yaml, why is this needed?

Comment (markxnelson) [reply THIRD_PARTY_LICENSES.txt:2314]: its transitive

File: application-operator/Makefile [hunk 2]
```diff
@@ -145,10 +145,6 @@ endif
 unit-test: go-install
 	$(GO) test -v ./apis/... ./controllers/... ./internal/... ./mcagent/...
 
-.PHONY: coverage
```
Comment (pfmackin) [inline application-operator/Makefile:148]: how can i check coverage for app operator alone if this is removed

Comment (markxnelson) [reply application-operator/Makefile:148]: cd application-operator; make coverage

Comment (markxnelson) [reply application-operator/Makefile:148]: ah i think i broke it :) i'll fixed

Comment (pfmackin) [reply application-operator/Makefile:148]: because if you don't check and you break it then you fail the build (waste of time) in your branch

File: THIRD_PARTY_LICENSES.txt (hunk 1 above)
Comment (pfmackin) [reply THIRD_PARTY_LICENSES.txt:2314]: i would have expected to see indirect

Comment (markxnelson) [reply THIRD_PARTY_LICENSES.txt:2314]: its not in the go.mod ???

File: application-operator/Makefile (hunk 2 above)
Comment (markxnelson) [reply application-operator/Makefile:148]: i fixed it for you

File: THIRD_PARTY_LICENSES.txt (hunk 1 above)
Comment (pfmackin) [reply THIRD_PARTY_LICENSES.txt:2314]: duhh right

**Review by pfmackin — APPROVED** [review]
